src/BSON/Binary.c

```diff
@@ -387,6 +388,27 @@ phongo_create_object_retval php_phongo_binary_create_object(zend_class_entry *cl
 #endif
 } /* }}} */
 
+static int php_phongo_binary_compare_objects(zval *o1, zval *o2 TSRMLS_DC) /* {{{ */
+{
+	php_phongo_binary_t *intern1, *intern2;
+	int retval;
+
+	intern1 = Z_BINARY_OBJ_P(o1);
+	intern2 = Z_BINARY_OBJ_P(o2);
+
+	retval = zend_binary_strcmp(intern1->data, intern1->data_len, intern2->data, intern2->data_len);
```

Comment:
This logic is incorrect. According to https://docs.mongodb.com/v3.2/reference/bson-types/#comparison-sort-order:

> MongoDB sorts BinData in the following order:
>
> * First, the length or size of the data.
> * Then, by the BSON one-byte subtype.
> * Finally, by the data, performing a byte-by-byte comparison.

Example:

```
> db.foo.find().sort({x:1})
{ "_id" : ObjectId("583bf6c1d0ece6c23583cf15"), "x" : BinData(0,"YmFy") }
{ "_id" : ObjectId("583bf5d8d0ece6c23583cf11"), "x" : BinData(0,"Zm9v") }
{ "_id" : ObjectId("583bf6c4d0ece6c23583cf16"), "x" : BinData(1,"YmFy") }
{ "_id" : ObjectId("583bf5e5d0ece6c23583cf12"), "x" : BinData(1,"Zm9v") }
{ "_id" : ObjectId("583bf5ecd0ece6c23583cf13"), "x" : BinData(0,"Zm9vZA==") }
{ "_id" : ObjectId("583bf5efd0ece6c23583cf14"), "x" : BinData(1,"Zm9vZA==") }
```